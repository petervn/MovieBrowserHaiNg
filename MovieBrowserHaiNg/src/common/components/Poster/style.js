import {StyleSheet} from 'react-native';
const style = StyleSheet.create({
  movieDetailPhoto: {},
  contentContainer: {
    flex: 1,
    opacity: 0.8,
    justifyContent: 'flex-end',
    padding: 15,
  },
  movieTitle: {
    fontSize: 16,
    color: 'white',
    fontWeight: 'bold',
  },
});

export default style;
