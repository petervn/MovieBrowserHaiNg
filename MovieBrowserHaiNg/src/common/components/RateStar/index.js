import React from 'react';
import {View, Image} from 'react-native';
import StartIcon from '../../asset/images/star.png';
import style from './style';

export default RateStar = ({voteAverage = 0}) => {
  if (isNaN(voteAverage)) {
    return null;
  }
  const average = Math.round(voteAverage);
  const length = Math.round(average * 0.5);

  return (
    <View style={style.container}>
      {[...Array(length)].map((e, i) => (
        <Image key={i} style={{height: 18, width: 18}} source={StartIcon} />
      ))}
    </View>
  );
};
