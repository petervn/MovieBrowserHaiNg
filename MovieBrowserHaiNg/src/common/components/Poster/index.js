import React from 'react';
import {View, Text, Image, ImageBackground} from 'react-native';
import LinearGradient from 'react-native-linear-gradient';

import {getImage} from '../../../common/utils';
import RateStar from '../RateStar';
import {black} from '../../styles/color';
import style from './style';

const Poster = ({title, backdropPath, voteAverage}) => (
  <View>
    <ImageBackground
      style={{
        width: '100%',
        height: 300,
        backgroundColor: 'transparent',
      }}
      source={getImage(backdropPath)}
      resizeMode="cover">
      <LinearGradient
        style={style.contentContainer}
        colors={[black, 'rgba(0, 0, 0, 0.14)']}>
        <View style={style.movieDetailPhoto}>
          <Text numberOfLines={3} style={style.movieTitle}>
            {title}
          </Text>
          <RateStar voteAverage={voteAverage} />
        </View>
      </LinearGradient>
    </ImageBackground>
  </View>
);

export default Poster;
