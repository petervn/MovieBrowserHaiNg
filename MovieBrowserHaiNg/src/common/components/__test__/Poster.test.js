import React from 'react';
import {Text, ImageBackground} from 'react-native';
import RateStar from '../RateStar';
import {shallow, mount} from 'enzyme';
import {expect as chaiExpect} from 'chai';
import Poster from '../Poster';

describe('Poster', () => {
  const title = 'Poster',
    backdropPath = 'invalidBackdropPath',
    voteAverage = 5;

  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = shallow(
        <Poster
          title={title}
          backdropPath={backdropPath}
          voteAverage={voteAverage}
        />,
      );
      expect(component).toMatchSnapshot();
    });

    it('should render image', () => {
      let component = mount(
        <Poster
          title={title}
          backdropPath={backdropPath}
          voteAverage={voteAverage}
        />,
      );
      const imageBackground = component.find(ImageBackground);
      chaiExpect(imageBackground).to.have.lengthOf(1);
      const uri = imageBackground.props().source.uri;
      expect(uri).toEqual('https://image.tmdb.org/t/p/w500invalidBackdropPath');
      const textValue = component.find(Text).text();
      expect(textValue).toEqual(title);
    });

    it('should render title and starts', () => {
      let component = mount(
        <Poster
          title={title}
          backdropPath={backdropPath}
          voteAverage={voteAverage}
        />,
      );
      chaiExpect(component.find(Text)).to.have.lengthOf(1);
      chaiExpect(component.find(RateStar)).to.have.lengthOf(1);
      const textValue = component.find(Text).text();
      expect(textValue).toEqual(title);
    });
  });
});
