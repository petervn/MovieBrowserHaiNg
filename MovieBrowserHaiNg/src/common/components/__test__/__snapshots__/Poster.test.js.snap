// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poster Rendering should match to snapshot 1`] = `
<View>
  <ImageBackground
    resizeMode="cover"
    source={
      Object {
        "uri": "https://image.tmdb.org/t/p/w500invalidBackdropPath",
      }
    }
    style={
      Object {
        "backgroundColor": "transparent",
        "height": 300,
        "width": "100%",
      }
    }
  >
    <LinearGradient
      colors={
        Array [
          "#000",
          "rgba(0, 0, 0, 0.14)",
        ]
      }
      end={
        Object {
          "x": 0.5,
          "y": 1,
        }
      }
      start={
        Object {
          "x": 0.5,
          "y": 0,
        }
      }
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-end",
          "opacity": 0.8,
          "padding": 15,
        }
      }
    >
      <View
        style={Object {}}
      >
        <Text
          numberOfLines={3}
          style={
            Object {
              "color": "white",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Poster
        </Text>
        <RateStar
          voteAverage={5}
        />
      </View>
    </LinearGradient>
  </ImageBackground>
</View>
`;
